### PR TITLE
fix/main.go

### LINKED ISSUE #103: Changes in Backend main.go code
hey @PradyumnaKrishna hope you remember me. As you said, I would like to make some changes in main.go code.

1) I can Enhanced error handling by checking and logging errors where necessary. Can Add logging statements to important sections to aid in debugging.
2) I will Update the JSONResponse function to handle JSON marshaling errors and return an internal server error response in such cases.
3) Can use a defer statement to ensure that the server is closed even if there's an error during initialization.
4) Used http.StatusOK instead of hardcoding the status code "200" in the response functions.
5) Ensured consistent formatting for better readability.

Also i would like to mention that codepeak'23 is going to end on 28th december so please can you assign me these task as soon as possible your early response will be appreciated. Thank You

### PR DESCRIPTION
**Pull request by itsmeabhishek03**
Close #103

hey @PradyumnaKrishna I have Added all changes as mentioned in issue #103. Please review it.

### CODE REVIEW

File: server/go/main.go [hunk 1]
```diff
@@ -146,14 +151,18 @@ func Init() {
 			log.Fatalf("socketio listen error: %s\n", err)
 		}
 	}()
-	defer server.Close()
-
-	http.Handle("/socket.io/", server)
-	http.Handle("/", myRouter)
```
Comment (PradyumnaKrishna) [inline server/go/main.go:152]: You've removed this which breaks the API.

**Review by PradyumnaKrishna — COMMENTED** [review]
LGTM only one minor issue.

**Comment by itsmeabhishek03** [comment]
> LGTM only one minor issue.

working on it

**Comment by itsmeabhishek03** [comment]
hey @PradyumnaKrishna sorry for the issue. Can you please check now.

**Review by PradyumnaKrishna — APPROVED** [review]
Thanks @itsmeabhishek03 for your contributions.

**Pull request merged**